pkgs/build-support/fetchsrht/default.nix

```diff
@@ -0,0 +1,29 @@
+{ fetchzip, lib }:
+
+{ user ? null
+, group ? null
+, owner ? if user != null then "~${user}" else throw "Please supply user or owner"
+, repo, rev
+, domain ? "sr.ht"
+, vc ? "git"
+, name ? "source"
+, ... # For hash agility
+} @ args:
+
+with lib;
+
+if !(elem vc [ "git" "hg" ])
+  then throw "fetchFromSourcehut: Version control ${vc} is not supported"
+  else assert true;
+
+let
+  baseUrl = "https://${vc}.${domain}/${owner}/${repo}";
```

Comment:
Are custom instances *guaranteed* to be set up in a way that this works?

Reply:
I have no idea, but I feel like it doesn't really matter if some instance aren't set up in that way since there is `fetchgit`.